cmd/basecli/counter/counter.go

```diff
@@ -12,68 +12,72 @@ import (
 	btypes "github.com/tendermint/basecoin/types"
 )
 
+//CounterTxCmd is the CLI command to execute the counter
+//  through the appTx Command
 var CounterTxCmd = &cobra.Command{
 	Use:   "counter",
 	Short: "add a vote to the counter",
 	Long: `Add a vote to the counter.
 
 You must pass --valid for it to count and the countfee will be added to the counter.`,
-	RunE: doCounterTx,
+	RunE: counterTxCmd,
 }
 
 const (
-	CountFeeFlag = "countfee"
-	ValidFlag    = "valid"
+	flagCountFee = "countfee"
+	flagValid    = "valid"
 )
 
 func init() {
 	fs := CounterTxCmd.Flags()
 	bcmd.AddAppTxFlags(fs)
-	fs.String(CountFeeFlag, "", "Coins to send in the format <amt><coin>,<amt><coin>...")
-	fs.Bool(ValidFlag, false, "Is count valid?")
+	fs.String(flagCountFee, "", "Coins to send in the format <amt><coin>,<amt><coin>...")
+	fs.Bool(flagValid, false, "Is count valid?")
 }
 
-func doCounterTx(cmd *cobra.Command, args []string) error {
-	tx := new(btypes.AppTx)
+func counterTxCmd(cmd *cobra.Command, args []string) error {
 	// Note: we don't support loading apptx from json currently, so skip that
 
-	// read the standard flags
-	err := bcmd.ReadAppTxFlags(tx)
+	// Read the app-specific flags
+	name, data, err := getAppData()
 	if err != nil {
 		return err
 	}
 
-	// now read the app-specific flags
-	err = readCounterFlags(tx)
+	// Read the standard app-tx flags
```

Comment:
I had actually originally set this up to go Cr4zY with the helper functions, but I decided that it makes sense to have some of the process still encapsulated in the application command - This is specifically in reference to the many variations of process flow I envisioned. There are really 4 elements here:
 - Get standard flags
 - Get plugin-specific flags
 - Create/Broadcast AppTx 
 - Output
Already while programming trackotron I've noticed how the generating the plugin-specific flags can require niche information from processed standard flags, I supposed we can pass in all the information in the form of a semi-complete AppTx - but this seems messy and unnecessary. In more complex (maybe multi-transaction) commands, some of these variables may be reused or reassigned before added to an AppTx and broadcast, and lastly a command may want to use different output. 
This tiny bit more verboseness ultimately makes the code much more flexible/clear